services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchValidationSpec.groovy

```diff
@@ -0,0 +1,601 @@
+            }
+        }
+
+        and: "The rest fields of 'meter' section are empty"
+        [srcSwitchValidateInfo, dstSwitchValidateInfo].each {
+            verifyMeterSectionsAreEmpty(it, ["missing", "proper", "excess"])
+        }
+
+        and: "Created rules are still stored in the 'proper' section"
+        def createdCookies = srcSwitchCreatedCookies + dstSwitchCreatedCookies
+        [srcSwitchValidateInfo, dstSwitchValidateInfo].each {
+            it.rules.proper.containsAll(createdCookies)
+            verifyRuleSectionsAreEmpty(it, ["missing", "excess"])
+        }
+
+        when: "Restore correct bandwidth via DB"
+        database.updateFlowBandwidth(flow.id, flow.maximumBandwidth)
+
+        then: "Misconfigured meters are moved into the 'proper' section"
+        def srcSwitchValidateInfoRestored = northbound.switchValidate(srcSwitch.dpId)
+        def dstSwitchValidateInfoRestored = northbound.switchValidate(dstSwitch.dpId)
+
+        srcSwitchValidateInfoRestored.meters.proper*.meterId.containsAll(srcSwitchCreatedMeterIds)
+        dstSwitchValidateInfoRestored.meters.proper*.meterId.containsAll(dstSwitchCreatedMeterIds)
+
+        [srcSwitchValidateInfoRestored, dstSwitchValidateInfoRestored].each {
+            verifyMeterSectionsAreEmpty(it, ["missing", "misconfigured", "excess"])
+        }
+
+        when: "Delete the flow"
+        flowHelper.deleteFlow(flow.id)
+
+        then: "Check that the switch validate request returns empty sections"
+        Wrappers.wait(WAIT_OFFSET) {
+            def srcSwitchValidateInfoAfterDelete = northbound.switchValidate(srcSwitch.dpId)
+            def dstSwitchValidateInfoAfterDelete = northbound.switchValidate(dstSwitch.dpId)
+            [srcSwitchValidateInfoAfterDelete, dstSwitchValidateInfoAfterDelete].each {
+                verifyRuleSectionsAreEmpty(it)
+                verifyMeterSectionsAreEmpty(it)
+            }
+        }
+    }
+
+    def "Switch validation is able to detect meter info into the 'missing' section"() {
+        when: "Create a flow"
+        def (Switch srcSwitch, Switch dstSwitch) = topology.activeSwitches
+        def flow = flowHelper.addFlow(flowHelper.randomFlow(srcSwitch, dstSwitch))
+        def srcSwitchCreatedMeterIds = getCreatedMeterIds(srcSwitch.dpId)
+        def dstSwitchCreatedMeterIds = getCreatedMeterIds(dstSwitch.dpId)
+
+        and: "Remove created meter on the srcSwitch"
+        northbound.deleteMeter(srcSwitch.dpId, srcSwitchCreatedMeterIds[0])
+
+        then: "Meters info/rules are moved into the 'missing' section on the srcSwitch only"
+        def srcSwitchCreatedCookies = getCookiesWithMeter(srcSwitch.dpId)
+        def dstSwitchCreatedCookies = getCookiesWithMeter(dstSwitch.dpId)
+
+        def srcSwitchValidateInfo = northbound.switchValidate(srcSwitch.dpId)
+        srcSwitchValidateInfo.rules.missing.size() == 1
+        srcSwitchValidateInfo.rules.missing.containsAll(srcSwitchCreatedCookies)
```

Comment:
the above two lines can be replaced with
`srcSwitchValidateInfo.rules.missing == srcSwitchCreatedCookies`

Reply:
I just want to make visible that one cookie is stored in the specific section.
I leave it as is.

Reply:
ok